File: programmers.co.kr/greedy/1.py
# https://programmers.co.kr/learn/courses/30/lessons/42862
# Gym Clothes

def solution(n, lost, reserve):
    ans = set()
    lostCopy = [l for l in lost if l not in reserve]
    reserve = [r for r in reserve if r not in lost]
    lost = lostCopy

    # Go in order asking for pair of extra clothes from neighbor
    for lostPerson in lost.copy():
        if (lostPerson-1) in reserve:
            lost.remove(lostPerson)
            reserve.remove(lostPerson-1)
        elif (lostPerson+1) in reserve:
            lost.remove(lostPerson)
            reserve.remove(lostPerson+1)
    return n - len(lost)

File: programmers.co.kr/DFS_BFS/1.cpp
// https://programmers.co.kr/learn/courses/30/lessons/43165
// Target Number

#include <string>
#include <vector>

using namespace std;

int search(int k, const vector<int> &numbers, int sum, const int target) {
    if (k == numbers.size()) {
        return (int)(sum == target);
    }
    // Addition case
    int ans = search(k+1, numbers, sum+numbers[k], target);
    ans += search(k+1, numbers, sum-numbers[k], target);
    return ans;
}

int solution(vector<int> numbers, int target) {
    return search(0, numbers, 0, target);
}
